src/components/ModelCamera/components/Camera.tsx

```diff
@@ -0,0 +1,118 @@
+
+import React, { FunctionComponent, useEffect, useRef } from 'react'
+import { Models, Alphabet } from "../../../types/Models";
+import styles from "./Camera.module.css";
+import * as Hands from "@mediapipe/hands";
+import Webcam from "react-webcam";
+import * as Cam from "@mediapipe/camera_utils";
+import { drawConnectors, drawLandmarks } from "@mediapipe/drawing_utils";
+
+interface CameraProps {
+    models: Models;
+}
+const Camera: FunctionComponent<CameraProps> = (props) => {
+
+    const webcamRef = useRef<Webcam>(null);
+    const canvasRef = useRef<HTMLCanvasElement>(null);
+    let camera = null;
+
+    async function onResults(results: any) {
+        let width = 0;
+        let height = 0;
+        if (webcamRef.current !== null && webcamRef.current.video !== null) {
+            width = webcamRef.current.video.videoWidth;
+            height = webcamRef.current.video.videoHeight;
+        }
+
+        if (canvasRef.current !== null) {
+            canvasRef.current.width = width;
+            canvasRef.current.height = height;
+        }
+
+        const canvasElement: HTMLCanvasElement | null = canvasRef.current;
+        const canvasCtx: CanvasRenderingContext2D | null | undefined =
+            canvasElement?.getContext("2d");
+
+        canvasCtx?.save();
+        canvasCtx?.clearRect(0, 0, canvasElement?.width!, canvasElement?.height!);
+
+        canvasCtx?.drawImage(
+            results.image,
+            0,
+            0,
+            canvasElement?.width!,
+            canvasElement?.height!
+        );
+
+        if (results.multiHandLandmarks && results.multiHandedness) {
+            for (let index = 0; index < results.multiHandLandmarks.length; index++) {
+                const classification = results.multiHandedness[index];
+                const isRightHand = classification.label === "Right";
+                const landmarks = results.multiHandLandmarks[index];
+                //@ts-ignore
+                drawConnectors(canvasCtx!, landmarks, Hands.HAND_CONNECTIONS, {
+                    color: "#4D4CAC",
+                    lineWidth: 5,
+                });
+                // @ts-ignore
+                drawLandmarks(canvasCtx, landmarks, {
+                    color: "#6f6dec",
+                    lineWidth: 2,
+                });
+            }
+        }
+        canvasCtx?.restore()
+    }
+
+    useEffect(() => {
+        const hands: Hands.Hands = new Hands.Hands({
```

Comment:
I think part of this, at least the instantiation of hands could be moved to a getStaticProps function. This would mean that hands it preloaded and ready to use when entering the page. I think the camera can't be moved to getStatic props because it's a client side thing that needs to happen.

Reply:
I think we can do that. Although, like I mentioned above, the camera component is CSR. Which I think it means that everything inside it will be CSR. Correct me if I'm wrong

Reply:
Actually nevermind, looking further I misread, there's no static prop being loaded here.